src/main/java/org/example/camunda/process/solution/service/TaskListService.java

```diff
@@ -121,6 +160,26 @@ private Task convert(io.camunda.tasklist.dto.Task task) {
         result.getVariables().put(var.getName(), var.getValue());
       }
     }
+
+    String schema = null;
+    String formKey = task.getFormKey();
+    try {
+      if (formKey.startsWith("camunda-forms:bpmn:")) {
+        String formId = formKey.substring(formKey.lastIndexOf(":") + 1);
+        // TODO: anyway to get process id from task??
+        String bpmnProcessId = task.getProcessName();
+        org.example.camunda.process.solution.facade.dto.Form form = null;
+        form = formService.findFormFromBpmnFile(bpmnProcessId + ".bpmn", formId);
+        schema = form.getSchema().toString();
+      } else {
+        org.example.camunda.process.solution.facade.dto.Form form =
+            formService.findFormJsonFileByFormKey(formKey);
+        schema = form.getSchema().toString();
+      }
+    } catch (Exception e) {
+      // can't find schema??
+    }
+    result.setFormSchema(schema);
```

Comment:
Hi @upgradingdave ,

Loading form schema for every task could be very consuming. And the user will probably not open all tasks. So I would rather load schema on demand. What do you think ?

Reply:
Hi @chDame. 

Yes, feel free to separate this into a separate method. I put this code here as convenience, but I meant to refine this for sure. 

For the UserTaskWorker, I found it easier to send both the task meta data + form schema over the web socket a the same time rather than having the client subscribe to 2 web socket topics; one for task list info and one for form schema. 

If the logic is separated into two functions, then the UserTaskWorker can just call both functions, one function to get task meta data and another to get form schema, and then send both over the websocket at once?

Happy to discuss more tomorrow if you want?